### PR TITLE
add data_source castai_eks_settings & eks iam example

### PR DESCRIPTION
**Pull request by ernestas2k**

### CODE REVIEW

File: examples/eks/versions.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    castai     = {
+      source  = "castai/castai"
+      version = "0.0.0-local"
```
Comment (ernestas2k) [inline examples/eks/versions.tf:5]: remove the explicit version from the example (we use it only in localdev example) as it isn't going to work for theprovider users. Users will download latest public version from registry.

File: examples/eks/iam.tf [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# IAM user required for CAST.AI
+
+provider "castai" {
+  api_url   = "https://console.cast.ai//"
```
Comment (ernestas2k) [inline examples/eks/iam.tf:4]: remove this url (our provider defaults to public api). By the way, api is located at https://api.cast.ai

**Comment by ernestas2k** [comment]
Looks good! Just few nitpicks. Let's merge this PR and further incorporate it in full eks cluster example 👍

File: examples/eks/iam.tf [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+provider "aws" {
+  region  = var.region
+}
+
+locals {
+  iam_user = "castai-eks-${var.cluster_name}"
+  aws_account_id = data.aws_caller_identity.current.account_id
+  instance_profile = "cast-${substr(var.cluster_name,0,40)}-eks-${substr(var.cluster_id,0,8)}"
+  instance_profile_roles = toset([
```
Comment (ernestas2k) [inline examples/eks/iam.tf:15]: Worth extracting the roles lists into the same datasource

Comment (matas-cast) [reply examples/eks/iam.tf:15]: agreed

**Comment by ernestas2k** [comment]
Approved! 🎆

**Review by matas-cast — APPROVED** [review]

**Pull request merged**